services/plan_manager.js

```diff
@@ -25,19 +26,35 @@ exports.post_process_create = async (subscription_plan,resp) => {
 
 
 exports.pre_process_get_plan_by_batch_id = async (req) => {
+    console.log("called")
     if (req.query == null ||  req.query.batch_id == null) {
         throw new Api400Error("Batch ID Not Provided")
     }
-    return req.query.batch_id
+    return {user_id : req.user.user_id , batch_id :req.query.batch_id}
 }
 
-exports.process_get_plan_by_batch_id = async (batch_id) => {
-    const plan = await models.SubscriptionPlan.findAll({
+exports.process_get_plan_by_batch_id = async (input_data) => {
+    const { user_id, batch_id } = input_data
+    const enrollment = await models.Enrollment.findOne({
         where: {
-            batch_id : batch_id
+            user_id: user_id
+        }
+    })
+    if (enrollment == null) {
+        return await models.SubscriptionPlan.findAll({
+            where: {
+                batch_id : batch_id
+            }
+        })
+    }
+    return  await models.SubscriptionPlan.findAll({
+        where: {
+            batch_id: batch_id,
+            type: {
+                [Op.notIn]: ["Demo", "demo", "DEMO"]
```

Comment:
filtering should be on price i.e. if price is not 0  or price > 0.if  type is entered as trial this will fail.